vfs/dir.go

```diff
@@ -188,13 +188,24 @@ func (d *Dir) invalidateDir(absPath string) {
 //
 // if entryType is a directory it invalidates the parent of the directory too.
 func (d *Dir) changeNotify(relativePath string, entryType fs.EntryType) {
-	defer log.Trace(d.path, "relativePath=%q, type=%v", relativePath, entryType)("")
 	d.mu.RLock()
 	absPath := path.Join(d.path, relativePath)
 	d.mu.RUnlock()
-	d.invalidateDir(vfscommon.FindParent(absPath))
+
+	defer log.Trace(d.path, "relativePath=%q, absPath=%q, type=%v", relativePath, absPath,
+		entryType)("")
+
+	/*
+		TODO: Check why this is necessary. Ideally, this should happen only if the
```

Comment:
I think it is because if you remove a file or a directory then you need to flush the parent of that directory.

The changes mechanism isn't descriptive enough to say whether we are deleting a file or dir, only that stuff happened on it.

Reply:
The Changes API is descriptive enough to not require flushing the parent since we calculate the parent anyway which in turn flushes the parent of the parent, which seems like a waste of resources.

How do you think this should be handled then to avoid issues with the other backends using the current implementation? Perhaps some argument to not invalidate the parent?